### PR TITLE
[FIX] qweb: fix t-if and t-else and add test

### PR DESCRIPTION
**Pull request by Gorash**
Can not use remove with the loop for in (in _compile_directive_content)

### CODE REVIEW

**Comment by Gorash** [comment]
@odony voilà

File: openerp/addons/base/tests/test_qweb.py [hunk 1]
```diff
@@ -85,6 +86,12 @@ def __call__(self, name, options):
                 arch = etree.tostring(root, encoding='utf-8', xml_declaration=True)
                 return arch
 
+def stringify_children(node):
+    parts = ([node.text] +
+            list(chain(*([c.text, etree.tostring(c), c.tail] for c in node.getchildren() if not isinstance(c, etree._Comment)))) +
+            [node.tail])
+    # filter removes possible Nones in texts and tails
+    return ''.join(filter(None, parts))
```
Comment (odony) [inline openerp/addons/base/tests/test_qweb.py]: I think we should be able to avoid this.. I'll have to manually test it, though.

Comment (xmo-odoo) [inline openerp/addons/base/tests/test_qweb.py]: Also the `parts` thing should probably use chain.from_iterable and ifilter, there's no reason to reify a list until the final join call.

Comment (Gorash) [inline openerp/addons/base/tests/test_qweb.py]: Shit, I had not had the email notification for the comment and as you told me it's good, I merged. xmo, I do not see what you mean :-/

Comment (xmo-odoo) [inline openerp/addons/base/tests/test_qweb.py]: line 90 creates a list, line 91 unpacks a generator into a possibly huge tuple (of lists) and creates a list, line 92 creates a list, lines 90~92 then create a new list from the previous 3, and line 94 create a new (filtered) list from everything else.

All of the list creation except the inner one (of text items from non-comment nodes) should be avoidable.

**Pull request merged**